compiler/p/codegen/OMRCodeGenerator.cpp

```diff
@@ -280,8 +280,12 @@ OMR::Power::CodeGenerator::initialize()
 
    // Standard allows only offset multiple of 4
    // TODO: either improves the query to be size-based or gets the offset into a register
-   if (comp->target().is64Bit())
+   if (!comp->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P9) &&
+       !(comp->target().cpu.isBigEndian() && comp->target().cpu.is(OMR_PROCESSOR_PPC_P8))
+      )
+      {
       cg->setSupportsAlignedAccessOnly();
+      }
 
```

Comment:
yes, this is correct from alignment-performance perspective.

however, from the comment in code, i suspected there was/might be a functional concern (i.e.  ld/std instruction encoding can only take immediate(s) which are multiple of 4).  on the other hand, as i think about it, this probably is not a concern anymore (codegen already took care of it), such that the comment should be removed.  please check.

Reply:
It looks like codegen handles this.

Offset of 11: (gets put in a register)
```
0x77f351ba0120 000000a4 [    0x77f350bf18f0] 3800000b          131  li      gr0, 000000000000000B
0x77f351ba0124 000000a8 [    0x77f350bf17e0] 7c63002a          131  ldx     gr3, [gr3, gr0]         # SymRef  <array-shadow>[#248  Shadow] [flags 0x80000601 0x0 ]
```

Offset of 16: (offset uses directly)
```
0x75ffe08a011c 000000a0 [    0x75ffdb161610] e8630010          134  ld      gr3, [gr3, 16]          # SymRef  <array-shadow>[#248  Shadow] [flags 0x80000601 0x0 ]
```

I'll remove the comment.